broadlink/__init__.py

```diff
@@ -290,15 +289,20 @@ def send_packet(self, command, payload):
 
         start_time = time.time()
         with self.lock:
+            cs = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            cs.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
```

Comment:
A little bit of overhead here, especially when we send a sequence of packets. I don't think we need to worry about that for now, but a possible optimization would be to implement a context manager:

```python3
with device.connect():
    for packet in packets:
        device.send_data(packet)
```
Where device.connect() is a context manager that handles sockets and authentication.

Reply:
This is so pythonic. I would really like to have this option at the integration level :point_up: